### PR TITLE
Implement yielding loop while waiting for input

### PR DESCRIPTION
**Pull request by noobpwnftw**
Some MPI implementations use busy-wait pooling, which will turn MPI_Bcast into busy-wait loop, workaround with our own yielding loop.

### CODE REVIEW

**Comment by noobpwnftw** [comment]
One thing I have to mention in this commit is that I find it absurd that if I don't use `static int flag;` like everywhere else, but just `int flag;`, it will segfault on first input command.
I wonder what is the cause...

**Comment by snicolet** [comment]
@noobpwnftw static variables are allocated somewhere on the heap (like global variables), while local variables are allocated on the stack.

**Comment by noobpwnftw** [comment]
@snicolet Let me rephrase my problem:
`MPI_Test` is a test function that returns the status of a previous non-blocking request, which returns immediately.
If I use a stack local `int flag;` like everywhere else, it would crash. The solution is to make it static which I find hard to see why the behavior is different, it must be wrong one way or the other.
